src/ride/ride.c

```diff
@@ -995,7 +995,9 @@ void ride_remove_peeps(int rideIndex)
 			exitZ = ride->station_heights[stationIndex];
 			mapElement = ride_get_station_exit_element(ride, exitX, exitY, exitZ);
 
-			exitDirection = mapElement->type & 3;
+			if (mapElement != NULL) {
+				exitDirection = mapElement->type & 3;
+			}
```

Comment:
This is worse. Your now accessing completely invalid memory and silently continuing since `exitDirection` is set to 255. I would instead set it to 0.